File: front-end/src/app/common/order-item.ts
import { CartItem } from "./cart-item";
import { Product } from "./product";

export class OrderItem {

  imageUrl: string;
  name: string;
  quantity: number;
  unitPrice: number;
  productId: number;
  product: Product | null = null;

  constructor(cartItem: CartItem) {

    this.imageUrl = cartItem.imageUrl;
    this.quantity = cartItem.quantity;
    this.unitPrice = cartItem.unitPrice;
    this.productId = cartItem.id;
    this.name = cartItem.name;

  }
}

File: back-end/src/main/java/com/example/ecommercebackend/entity/OrderItem.java
package com.example.ecommercebackend.entity;


import jakarta.persistence.*;
import lombok.Data;

@Entity
@Table(name="order_item")
@Data
public class OrderItem {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private int id;

    @Column(name="image_url")
    private String imageUrl;

    @Column(name="quantity")
    private int quantity;

    @Column(name="name")
    private String name;

    @Column(name="unit_price")
    private int unitPrice;


    @OneToOne(fetch = FetchType.EAGER)
    private Product product;

    @ManyToOne
    @JoinColumn(name="order_id", nullable = false)
    private Order order;

    @Column(name="product_id", insertable = false, updatable = false)
    private Integer productId; // This field can be used for reference
}
